### PR TITLE
Explore: Remove storing derived "loading" property

### PR DESCRIPTION
**Pull request by ifrost**
`ExploreState.panes.[exploreId].loading` property is derived from `queryResponse.loadingState`. When a derived state property gets out of sync with `queryResponse` it causes issues with showing the correct loading state. It happened when fixing #70235. After fixing canceling the query, the state of `loading` property was updated (by cancelQueriesAction), but not `queryResponse.loadingState`. (https://github.com/grafana/grafana/pull/70235#issuecomment-1596889753). 

This PR attempts to remove that derived state in favor of using a simple selector.

### CODE REVIEW

**Comment by Elfo404** [comment]
love the selector approach to determine the loading state <3

File: public/app/features/explore/state/time.ts [hunk 1]
```diff
@@ -165,7 +165,6 @@ export const timeReducer = (state: ExploreItemState, action: AnyAction): Explore
       },
       isLive: live,
       isPaused: live ? false : state.isPaused,
-      loading: live,
```
Comment (ifrost) [inline public/app/features/explore/state/time.ts:168]: This was introduced in https://github.com/grafana/grafana/pull/18804 (reducers.ts file). Probably to clear loading state from logs panel when streaming is paused (in such case queryResponse.state is still `streaming`), but we don't have any loading state in logs panel while streaming, we may need to revisit it when switching to PanelChrome.

File: public/app/features/alerting/unified/components/rule-editor/VizWrapper.tsx [hunk 2]
```diff
@@ -37,7 +37,6 @@ export const VizWrapper = ({ data, thresholds, thresholdsType }: Props) => {
           <div style={{ width }}>
             {isTimeSeriesData ? (
               <GraphContainer
-                loading={data.state === LoadingState.Loading}
```
Comment (ifrost) [inline public/app/features/alerting/unified/components/rule-editor/VizWrapper.tsx:40]: This property is not being used, can be safely removed.

**Review by matyax — COMMENTED** [review]
Log changes look good 👍

**Review by gillesdemey — APPROVED** [review]
Alerting changes LGTM 👍

**Review by Elfo404 — APPROVED** [review]

**Pull request merged**